tempesta_fw/ss_skb.c

```diff
@@ -741,17 +753,29 @@ __skb_fragment(struct sk_buff *skb, char *pspt, int len, TfwStr *it)
 		return ret;
 	if ((it->ptr == NULL) || (it->skb == NULL))
 		return -EFAULT;
-	it->len = len;
 
 	/* Return the length of processed data. */
-	return abs(len);
+	return (it->len = abs(len));
```

Comment:
I wouldn't use an assignment in a `return()`. The syntax is correct, of course, but I find this practice doubtful.